source/_posts/2023-10-01-php-core-roundup-17.md

```diff
@@ -0,0 +1,122 @@
+
+## PHP Security Audit organized by The PHP Foundation 💜
+
+The PHP Foundation is planning to organized a security audit in PHP source code. Derick Rethans 💜 emailed the PHP Internals mailing list requesting opinions to identify the places in the PHP source code where checking this will have the most impact. Feel free to join the [conversation](https://externals.io/message/121135) if you have suggestions.
+
+## Releases
+
+The PHP development team released two new versions in September 2023:
+
+**[PHP 8.2.11](TODO)** and **[PHP 8.1.24](TODO)**
+
+These releases include several bug fixes and improvements, notably in areas such as Core, DOM, Iconv, Intl, MySQLnd, ODBC, SimpleXML, SPL, and SQLite3.
+
+## Recent RFCs and Mailing List Discussions
+
+
+> Changes and improvements to PHP are discussed, reported, and voted on by the PHP Foundation Team, the PHP development team, and contributors. Bug reports are made to the PHP [issue tracker](https://github.com/php/php-src/issues), changes are discussed in [mailing lists](https://www.php.net/mailing-lists.php), minor code changes are proposed as [pull requests](https://github.com/php/php-src/issues), and major changes are discussed in detail and voted on as [PHP RFCs](https://wiki.php.net/rfc). [Documentation](https://github.com/php/doc-en/) and the [php.net website](https://github.com/php/web-php) changes are also discussed and improved at their relevant Git repositories on GitHub.
+
+<br>
+Hundreds of awesome PHP contributors put their efforts into improvements to the PHP code base, documentation, and the php.net website. Here is a summary of some changes made by the people behind PHP. Things marked with 💜 are done by the PHP Foundation team.
+
+## RFC Updates
+
+### Declined: [Support optional suffix parameter in tempnam](https://wiki.php.net/rfc/tempnam-suffix-v2) by Athos Ribeiro
+
+RFC proposes to add a new optional suffix parameter to the `tempnam()` function.
+
+A suffix could provide even more semantic value or context for a user inspecting the generated files, and, in specific situations, could even provide more context for software processing such files. Right now, users can only add a prefix.
+
+### In Voting: [Increasing the default BCrypt cost](https://wiki.php.net/rfc/bcrypt_cost_2023) by Tim Düsterhus
+
+RFC proposes to increase default BCrypt cost, which denotes the algorithmic cost that should be used, from 10 to 11 (doubling the time) or 12 (quadrupling the time). The [RFC](https://wiki.php.net/rfc/bcrypt_cost_2023) and the relevant [mailing list thread](https://externals.io/message/121004) mention several benchmarks showing the execution time differences in various cost levels.
+
+### Under Discussion: [DOM HTML5 parsing and serialization](https://wiki.php.net/rfc/domdocument_html5_parser) by Niels Dossche
+
+RFC proposes to add a new class `DOM\HTML5Document` to dom extension. This will add support for HTML5 document parsing while keeping backwards compatibility.
```

Comment:
This description is not up to date with the latest developments. /cc @nielsdos

Reply:
Right, I'll make a suggestion when I am behind my desktop.

Reply:
I believe something like this works, if it's not too long:

RFC proposes to add two new classes: `DOM\HTMLDocument` and `DOM\XMLDocument` to the dom extension. Furthermore, existing dom classes in the global namespace get an alias in the new DOM namespace. The `HTMLDocument` class will add support for HTML5 document parsing and serializing while keeping backwards compatibility. The `XMLDocument` class serves as a modern alternative to `\DOMDocument`, which is retained for compatibility. These new classes also provide a more misuse-resistant API for loading documents.